climada/hazard/base.py

```diff
@@ -450,6 +452,60 @@ def sanitize_event_ids(self):
             LOGGER.debug('Resetting event_id.')
             self.event_id = np.arange(1, self.event_id.size + 1)
 
+    def local_return_period(self, threshold_intensities):
+        """Compute local return periods for given hazard intensities.
+    
+        Parameters
+        ----------
+        threshold_intensities : np.array
+            Hazard intensities to consider.
+    
+        Returns
+        -------
+        gdf : gpd.GeoDataFrame
+            GeoDataFrame containing return periods for given threshold intensity
+            description of units of values and threshold intensities in gdf.columns.name
+        """
+        #check frequency unit
+        if self.frequency_unit not in ['1/year', 'annual', '1/y', '1/a']:
+            LOGGER.warning("The Hazard's frequency unit is %s and not %s which "
+                "most likely leads to incorrect results",
+                self.frequency_unit, u_const.DEF_FREQ_UNIT)
+
+        # Ensure threshold_intensities is a numpy array
+        threshold_intensities = np.array(threshold_intensities)
+        
+        num_cen = self.intensity.shape[1]
+        return_periods = np.zeros((len(threshold_intensities), num_cen))
+        
+        # Process each centroid in chunks as in local_exceedance_inten
+        block_size = CONFIG.max_matrix_size.int() // self.intensity.shape[0]
+        if not block_size:
+            raise ValueError('Increase max_matrix_size configuration parameter to >'
+                             f'{self.intensity.shape[0]}')
+        
+        for start_col in range(0, num_cen, block_size):
+            end_col = min(start_col + block_size, num_cen)
+            return_periods[:, start_col:end_col] = self._loc_return_period(
+                threshold_intensities,
+                self.intensity[:, start_col:end_col].toarray()
+                )
```

Comment:
sorry, maybe I didn't understand what max_matrix_size represents. Is it the number of columns, the number of rows, or the (number cols) x (number rows). I thought it was the last one. 

I also think that the chunking is not needed, especially because I wanted to rewrite these functions using the general fit method discussed in issue #904. I though we'd just clean this up there, but if you say it's ok I can remove the chunking already here